setup.py

```diff
@@ -1,181 +0,0 @@
-        "Programming Language :: Python :: 3.8",
-        "Programming Language :: Python :: 3.9",
-        "Programming Language :: Python :: 3.10",
-        "Programming Language :: Python :: 3 :: Only",
-    ],
-    # This field adds keywords for your project which will appear on the
-    # project page. What does your project relate to?
-    #
-    # Note that this is a list of additional keywords, separated
-    # by commas, to be used to assist searching for the distribution in a
-    # larger catalog.
-    keywords="sample, setuptools, development",  # Optional
-    # When your source code is in a subdirectory under the project root, e.g.
-    # `src/`, it is necessary to specify the `package_dir` argument.
-    package_dir={"": "src"},  # Optional
-    # You can just specify package directories manually here if your project is
-    # simple. Or you can use find_packages().
-    #
-    # Alternatively, if you just want to distribute a single Python file, use
-    # the `py_modules` argument instead as follows, which will expect a file
-    # called `my_module.py` to exist:
-    #
-    #   py_modules=["my_module"],
-    #
-    packages=find_packages(where="src"),  # Required
-    # Specify which Python versions you support. In contrast to the
-    # 'Programming Language' classifiers above, 'pip install' will check this
-    # and refuse to install the project if the version does not match. See
-    # https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires
-    python_requires=">=3.7, <4",
-    # This field lists other packages that your project depends on to run.
-    # Any package you put here will be installed by pip when your project is
-    # installed, so they must be valid existing projects.
-    #
-    # For an analysis of "install_requires" vs pip's requirements files see:
-    # https://packaging.python.org/discussions/install-requires-vs-requirements/
-    install_requires=["peppercorn"],  # Optional
-    # List additional groups of dependencies here (e.g. development
-    # dependencies). Users will be able to install these using the "extras"
-    # syntax, for example:
-    #
-    #   $ pip install sampleproject[dev]
-    #
-    # Similar to `install_requires` above, these must be valid existing
-    # projects.
-    extras_require={  # Optional
-        "dev": ["check-manifest"],
-        "test": ["coverage"],
-    },
-    # If there are data files included in your packages that need to be
-    # installed, specify them here.
-    package_data={  # Optional
-        "sample": ["package_data.dat"],
-    },
-    # Although 'package_data' is the preferred approach, in some case you may
-    # need to place data files outside of your packages. See:
-    # http://docs.python.org/distutils/setupscript.html#installing-additional-files
-    #
-    # In this case, 'data_file' will be installed into '<sys.prefix>/my_data'
-    data_files=[("my_data", ["data/data_file"])],  # Optional
```

Comment:
Removing the inclusion of `data/data_file` (sorry, I appear not to be able to comment on the removed file itself) seems independent of switching to `pyproject.toml`. I'm fine with *not* showing an example of non-package data, as that's not recommended practice these days, but it should be in a separate PR.

Reply:
@abravalheri How should this be represented in `pyproject.toml` - does it need to be a list of lists of lists?

```toml
[tool.setuptools]
data-files = [
  ["my_data", ["data/data_file"]],
]
```

Reply:
There's some previous discussion about this [here](https://github.com/pypa/sampleproject/pull/166#issuecomment-1079727615) and [here](https://github.com/pypa/sampleproject/pull/166#issuecomment-1080895022). IIRC this needs https://github.com/pypa/setuptools/issues/3191 in order to work?

Reply:
Frankly if setuptools maintainers have reservations about allowing this functionality (as https://github.com/pypa/setuptools/issues/3191 suggests) we'd probably be better just removing it as an example altogether.